src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -128,9 +132,47 @@ class ClusterTaskManager {
   /// \param[in] any_pending True if there's any pending exemplar.
   /// \return True if any progress is any tasks are pending.
   bool AnyPendingTasks(Task *exemplar, bool *any_pending, int *num_pending_actor_creation,
-                       int *num_pending_tasks) const;
+                       int *num_pending_tasks) const override;
 
-  std::string DebugString() const;
+  /// Return the resources that were being used by this worker.
+  void FreeLocalTaskResources(std::shared_ptr<WorkerInterface> worker) override;
+
+  /// When direct call task is blocked, the worker who is executing the task should give
+  /// up the cpu resources allocated for the running task for the time being and the
+  /// worker itself should also be marked as blocked.
+  bool ReleaseCpuResourcesAndMarkWorkerAsBlocked(std::shared_ptr<WorkerInterface> worker,
+                                                 bool release_resources) override;
+
+  // When direct call task is unblocked, the cpu resources that the worker gave up should
+  // be returned to it.
+  bool ReturnCpuResourcesAndMarkWorkerAsUnblocked(
+      std::shared_ptr<WorkerInterface> worker) override;
+
+  // Schedule and dispatch tasks.
+  void ScheduleAndDispatchTasks() override;
+
+  /// Handle the resource usage updated event of the specified node.
+  ///
+  /// \param node_id ID of the node which resources are updated.
+  /// \param resource_data The node resources.
+  void OnNodeResourceUsageUpdated(const NodeID &node_id,
+                                  const rpc::ResourcesData &resource_data) override;
+
+  /// Handle the bundle resources prepared event.
+  void OnBundleResourcesPrepared() override;
+
+  /// Handle the bundle resources committed event.
+  void OnBundleResourcesCommitted() override;
+
+  /// Handle the reserved resources canceled event.
+  void OnReservedResourcesCanceled() override;
+
+  /// Handle the object missing event.
+  void OnObjectMissing(const ObjectID &object_id,
+                       const std::vector<TaskID> &waiting_task_ids) override;
```

Comment:
Can we remove the above four functions OnBundleResourcesPrepared, OnBundleResourcesCommitted, OnReservedResourcesCanceled, and OnObjectMissing? They seem to be no-ops or just calling ScheduleAndDispatch().

Reply:
Because there are some differences in the processing of these event you mentioned above between new and old schedulers in `NodeManager.cc`, so I add these functions to eliminate `if (new_scheduler_enabled_) {...} else {...}`.

For those event that should be ignored in new scheduling, what do you think if I implement them as empty in the `ClusterTaskManagerInterface`. @ericl @raulchen

Reply:
`OnBundleResourcesPrepared`, `OnBundleResourcesCommitted` and `OnReservedResourcesCanceled` could be removed based on @stephanie-wang 's suggestion, while `OnObjectMissing` doesn't seem to be able to be removed, I can add a `TODO` to remove this method once we remove the legacy scheduler.